### PR TITLE
blocks: move the `new` from the block `::make()` to core

### PR DESCRIPTION
**Pull request by ThomasHabets**
This PR is a preliminary one to deprecate the direct use of `get_initial_pointer`.

The reasons to change this centralizing the place where `new` is called are:
1. Ease of readability
2. For future magic-making it's easier if less logic is distributed (e.g. if `sptr_magic` need is removed, then it's easy to just change to `std::make_shared`)
3. Fewer steps for all blocks to perform by unifying creating the object and creating the shared pointer.

If you agree then I can extend this to all call sites.

### CODE REVIEW

**Comment by mbr0wn** [comment]
This will also require modtool changes.

**Comment by ThomasHabets** [comment]
Added a commit with modtool changes.

Curious. Modtool generates output that doesn't conform to style guide. Is it assumed it'll be `clang-format`ted?

**Comment by mbr0wn** [comment]
> Curious. Modtool generates output that doesn't conform to style guide. Is it assumed it'll be `clang-format`ted?

No -- that's a bug.

**Comment by ThomasHabets** [comment]
I'll fix the generated output in a separate PR.

**Comment by michaelld** [comment]
@ThomasHabets So is the intent here to do the change and show how it will be implemented via the null_source ... and, then once this is in place, you'll do another PR that changes all of the uses of `new` to `::make` as per the null_source? I'm just trying to understand why just the null_source is changed, and nothing else ... thx!

**Comment by ThomasHabets** [comment]
@michaelld that's right. That's what I meant with "If you agree then I can extend this to all call sites".

File: gnuradio-runtime/include/gnuradio/sptr_magic.h [hunk 1]
```diff
@@ -41,6 +45,17 @@ std::shared_ptr<T> get_initial_sptr(T* p)
     return std::dynamic_pointer_cast<T, gr::basic_block>(
         detail::sptr_magic::fetch_initial_sptr(p));
 }
+
+/*
+ * \brief GNURadio version of std::make_shared<> that also provides magic. For
```
Comment (mbr0wn) [inline gnuradio-runtime/include/gnuradio/sptr_magic.h:50]:
```suggestion
 * \brief GNU Radio version of std::make_shared<> that also provides magic. For
```

**Comment by ThomasHabets** [comment]
@mbr0wn So it should be safe to do the rest of the changes now?

**Comment by michaelld** [comment]
> @mbr0wn So it should be safe to do the rest of the changes now?

My vote is: yes!

**Comment by michaelld** [comment]
@marcusmueller should be the decider here if possible, since this will start the process of changing how `make` happens with blocks ... which is, like, pretty serious stuff!

**Comment by ThomasHabets** [comment]
By the power of `sed` I've changed all callsites.

Wasn't that much work after all.

**Comment by mormj** [comment]
would like to get signoff from @marcusmueller on this one

**Comment by ThomasHabets** [comment]
@marcusmueller ping?

**Comment by ThomasHabets** [comment]
@marcusmueller ping again?

**Comment by ThomasHabets** [comment]
@marcusmueller weekly ping :-)

**Comment by marcusmueller** [comment]
hey Thomas, sorry, somehow this was constantly slipping. 

So, let's merge in pybind11 and then this!

**Comment by ThomasHabets** [comment]
@marcusmueller looks like pybind11 was merged? Or are we still waiting for https://github.com/gnuradio/gnuradio/pull/3611 ?

Or were you waiting for me to resolve the conflicts? (I will, shortly)

**Comment by mormj** [comment]
I don't think this should be held up by #3611

**Comment by ThomasHabets** [comment]
Rebased.

**Comment by ThomasHabets** [comment]
@marcusmueller so this should be all set for merging?

**Comment by ThomasHabets** [comment]
ping?

**Pull request merged**